pkg/redis/redis.go

```diff
@@ -110,15 +116,34 @@ type FailoverConfig struct {
 	MasterName string   `name:"master-name" description:"Redis Sentinel master name"`
 }
 
-func dial(ctx context.Context, network, addr string) (net.Conn, error) {
+func (c Config) dial(ctx context.Context, network, addr string) (net.Conn, error) {
 	var timeout time.Duration
 	deadline, ok := ctx.Deadline()
 	if ok {
 		timeout = time.Until(deadline)
 	}
-	conn, err := net.DialTimeout(network, addr, timeout)
-	if err != nil {
-		return nil, err
+	var (
+		conn net.Conn
+		err  error
+	)
+	dialer := &net.Dialer{Timeout: timeout}
+	if c.TLS.Require {
+		tlsConfig := &tls.Config{
+			MinVersion: tls.VersionTLS12,
+		}
+		err := c.TLS.Client.Apply(tlsConfig)
+		if err != nil {
+			return nil, err
+		}
+		conn, err = tls.DialWithDialer(dialer, network, addr, tlsConfig)
+		if err != nil {
+			return nil, fmt.Errorf("Error connecting to %q: %w", addr, err)
```

Comment:
`redis: dial %q: %w`

Reply:
This shouldn't have been committed in the first place.